src/ui/inventory.rs

```diff
@@ -33,28 +32,235 @@ pub fn build_menu(api: &mut api::API, machine_index: i32) {
+                    _ => deny()
+                }
+              }
+              else {
+                deny();
+              }
+            },
+            KEY_LEFT => {
+              ui_common::destroy_win(win);
+              return;
+            },
+            _ => {
+              refresh();
+            }
+        }
+        
+        for n in 0..slot_count {
+          if n as i32 == selected_slot {
+              wattron(win, A_REVERSE());
+          }
+          if slots[n].empty {
+            wattron(win, COLOR_PAIR(1));
+          }
+          if !slots[n].active {
+            wattron(win, A_DIM());
+          }
+          mvwprintw(
+              win, 3 + n as i32, 2,
+              format!("{} ({} credits)", slots[n].name, slots[n].price).as_str(),
+          );
+          wattroff(win, A_DIM());
+          wattroff(win, COLOR_PAIR(1));
+          wattroff(win, A_REVERSE());
+        }
+
+        refresh();
+        wrefresh(win);
+
+        key = getch(); 
+      }
+    },
+    _ => {
+        endwin();
+        panic!("Error: Could not query inventory");
+    }
   }
+}
 
-  // TODO: Get real amt of credits.
-  mvwprintw(win, height - 3, width - 20, "Credits: 69420");
+pub fn vend() {
+  /* Get the screen bounds. */
+  let mut max_x = 0;
+  let mut max_y = 0;
+  getmaxyx(stdscr(), &mut max_y, &mut max_x);
 
+  /* Create the window */
+  let height = 5;
+  let width = 40;
+  let start_y = (max_y - height) / 2;
+  let start_x = (max_x - width) / 2;
+  let win = ui_common::create_win(start_y, start_x, height, width);
+
+  mvwprintw(win, 1, 3, "Item Dropped!");
+  mvwprintw(win, 3, 3, "Press any key to continue");
   wrefresh(win);
   refresh();
-  let requested_machine = getch();
-  //TODO: something. Drop drink I guess.
+  getch();
   ui_common::destroy_win(win);
 }
+
+pub fn deny() {
+  attron(COLOR_PAIR(1));
+  /* Get the screen bounds. */
+  let mut max_x = 0;
+  let mut max_y = 0;
+  getmaxyx(stdscr(), &mut max_y, &mut max_x);
+
+  /* Create the window */
+  let height = 5;
+  let width = 40;
+  let start_y = (max_y - height) / 2;
+  let start_x = (max_x - width) / 2;
+  let win = ui_common::create_win(start_y, start_x, height, width);
+
+  wattron(win, COLOR_PAIR(1));
+  mvwprintw(win, 1, 3, "Slot empty or insufficient funds.");
+  mvwprintw(win, 3, 3, "Press any key to continue");
+  wrefresh(win);
+  refresh();
+  getch();
+  wattroff(win, COLOR_PAIR(1));
+  ui_common::destroy_win(win);
+  attroff(COLOR_PAIR(1));
+}
+
+pub fn parse_inventory(
+  status: &Value,
+  machine_index: i32,
+) -> Result<Vec<Item>, Box<dyn std::error::Error>> {
+  // TODO: There's a better way to handle these. You could just
+  // Unwrap them, or do something else.
+  let drinks: &Map<String, Value> = match status.as_object() {
+    Some(drinks) => drinks,
+    None => panic!("Fuck"),
+  };
+
+  let machines: &Vec<Value> = match drinks["machines"].as_array() {
+    Some(machines) => machines,
+    None => panic!("Fuck"),
```

Comment:
So better way to handle these cases I've learned is with like https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or